cleverhans/experimental/certification/dual_formulation.py

```diff
@@ -154,35 +154,22 @@ def construct_lanczos_params(self):
     """
     # Using autograph to automatically handle
     # the control flow of minimum_eigen_vector
-    self.min_eigen_vec = autograph.to_graph(utils.lanczos_decomp)
+    self.min_eigen_vec = autograph.to_graph(utils.tf_lanczos_smallest_eigval)
 
     def _m_vector_prod_fn(x):
       return self.get_psd_product(x)
     def _h_vector_prod_fn(x):
       return self.get_h_product(x)
 
     # Construct nodes for computing eigenvalue of M
-    self.alpha_m, self.beta_m, self.Q_m = self.min_eigen_vec(_m_vector_prod_fn,
-                                                             0,
-                                                             self.matrix_m_dimension,
-                                                             self.lzs_params['min_iter'])
-
-    self.eig_max_placeholder = tf.placeholder(tf.float32, shape=[])
-    self.alpha_m_hat, self.beta_m_hat, self.Q_m_hat = self.min_eigen_vec(_m_vector_prod_fn,
-                                                                         self.eig_max_placeholder,
-                                                                         self.matrix_m_dimension,
-                                                                         self.lzs_params['max_iter'])
-
-    # Construct nodes for computing eigenvalue of H
-    self.alpha_h, self.beta_h, self.Q_h = self.min_eigen_vec(_h_vector_prod_fn,
-                                                             0,
-                                                             self.matrix_m_dimension-1,
-                                                             self.lzs_params['max_iter'])
-
-    self.alpha_h_hat, self.beta_h_hat, self.Q_h_hat = self.min_eigen_vec(_h_vector_prod_fn,
-                                                                         self.eig_max_placeholder,
-                                                                         self.matrix_m_dimension-1,
-                                                                         self.lzs_params['max_iter'])
+    self.m_min_eig, self.m_min_vec = self.min_eigen_vec(_m_vector_prod_fn,
+                                                        self.matrix_m_dimension,
+                                                        self.lzs_params['max_iter'],
+                                                        dtype=tf.float64)
```

Comment:
is using float64 helpful here?
I think if you want to alter dtype for lanczos, it would be good to actually have it stored in the class:
```
class DualFormulation(object):

  def __init__(...):
    # ...
    self.lzs_dtype = tf.float64  # dtype for Lanczos algorithm

  def construct_lanczos_params(...):
    # ...
    self.m_min_eig, self.m_min_vec = self.min_eigen_vec(
        _m_vector_prod_fn,
        self.matrix_m_dimension,
        self.lzs_params['max_iter'],
        dtype=self.lzs_dtype)
```
This way you we can easily switch between dtypes

Reply:
yes float64 is more precise